### PR TITLE
chore: improve debugging for no subnets error in task run

### PR DESCRIPTION
**Pull request by ota42y**
Improve error message to let users know that a tag is required for subnet to run task.

Please read this issue 
https://github.com/aws/copilot-cli/issues/2022

### CODE REVIEW

File: internal/pkg/task/env_runner.go
Comment (efekarakus) [inline internal/pkg/task/env_runner.go]: Can we revert this change? looks like an extra space was added at the end 🙏

File: internal/pkg/task/env_runner.go
Comment (efekarakus) [inline internal/pkg/task/env_runner.go]: Instead of having the error message contain a recommendation "(need %s and %s tags?)" can we use the `log` pkg to add additional context? So the error can be specific but logging can suggest additional actions.

```suggestion
		log.Warningf("Subnets need to be tagged with %s and %s. Are your subnets tagged?\n", deploy.AppTagKey, deploy.EnvTag.Key)
		return nil, errNoSubnetFound
```

Ideally, we would catch this error in the `cli` pkg and do the logging there since this package is meant to be a library pkg.

**Review by efekarakus — COMMENTED** [review]
Awesome thank you so much for making debugging this issue better 🙇 ! Some small suggestions below

File: internal/pkg/task/env_runner.go
Comment (ota42y) [reply internal/pkg/task/env_runner.go]: This is miss, sorry.

**Comment by ota42y** [comment]
I have completed the fix!
I changed to show a warning by error type in cli package so I made error struct public for error check.

File: internal/pkg/cli/task_run.go
Comment (efekarakus) [inline internal/pkg/cli/task_run.go]: nit: so that wrapped errors are also matched :)
```suggestion
		if errors.Is(err, task.ErrNoSubnetFound) {
```

File: internal/pkg/task/errors.go
Comment (efekarakus) [inline internal/pkg/task/errors.go:13]: Can we add a top-level comment here since we now have exported errors
```suggestion
// Errors returned while trying to run a task.
var (
	ErrNoSubnetFound = errors.New("no subnets found")
```

**Review by efekarakus — APPROVED** [review]
Yay this looks great! Thank you so much, I left some small nits. Once addressed, we'll remove the "do-not-merge" label which will make [@mergifyio](https://mergify.io/) automatically merge your PR 🚀

**Comment by ota42y** [comment]
I fixed it.

However, if you merge https://github.com/aws/copilot-cli/pull/2024 to next version, you must not merge this PR.
Because I changed to use Stack instead of tags in #2024 , so this warning message is no longer needed.
If you don't merge #2024 to next version, this message is useful for users.

**Comment by efekarakus** [comment]
Yay sounds good! We'll merge #2024 instead!

**Comment by ota42y** [comment]
Ok, I'm going to close this PR when #2024 merged

**Comment by ota42y** [comment]
#2024 is merged 🎉

**Pull request closed without merging**
